csharp/src/Google.Protobuf/JsonTokenizer.cs

```diff
@@ -565,6 +572,38 @@ private void PopContainer()
                 };
             }
 
+            private static StringBuilder AcquireStringBuilder(int capacity = DEFAULT_CAPACITY)
+            {
+                if (capacity <= MAX_BUILDER_SIZE)
+                {
+                    var sb = t_cachedInstance;
+                    // Avoid stringbuilder block fragmentation by getting a new StringBuilder
+                    // when the requested size is larger than the current capacity
+                    if (capacity <= sb?.Capacity)
+                    {
+                        t_cachedInstance = null;
+                        sb.Clear();
+                        return sb;
+                    }
+                }
+                return new StringBuilder(capacity);
+            }
+
+            private static void Release(StringBuilder sb)
```

Comment:
I don't think we need this as a separate method, given that we only ever call it from GetStringAndRelease.